src/main/java/games/strategy/triplea/attachments/TriggerAttachment.java

```diff
@@ -112,19 +112,19 @@ public static TriggerAttachment get(final PlayerID player, final String nameOfAt
       if (playersToSearch == null) {
         throw new IllegalStateException(
             "Triggers: No trigger attachment for:" + player.getName() + " with name: " + nameOfAttachment);
-      } else {
-        for (final PlayerID otherPlayer : playersToSearch) {
-          if (otherPlayer == player) {
-            continue;
-          }
-          ta = (TriggerAttachment) otherPlayer.getAttachment(nameOfAttachment);
-          if (ta != null) {
-            return ta;
-          }
+      }
+
+      for (final PlayerID otherPlayer : playersToSearch) {
+        if (otherPlayer == player) {
+          continue;
+        }
+        ta = (TriggerAttachment) otherPlayer.getAttachment(nameOfAttachment);
+        if (ta != null) {
+          return ta;
         }
-        throw new IllegalStateException(
-            "Triggers: No trigger attachment for:" + player.getName() + " with name: " + nameOfAttachment);
       }
+      throw new IllegalStateException(
+          "Triggers: No trigger attachment for:" + player.getName() + " with name: " + nameOfAttachment);
```

Comment:
```java
ta = playersToSearch.stream()
  .filter(o ->! Objects.equals(o, player)// Prefering equals over reference equality here, not sure if this is safe.
  .map(o -> o.getAttachment(nameOfAttachment))
  .map(TriggerAttachment.class::cast)
  .filter(Objects::nonNull)
  .findAny()
  .orElseThrow(() -> new IllegalStateException("Error Text");
return ta;
```
Or something along those lines.

Reply:
Probably will skip this one for now because these uses of reference equality being flagged by Error Prone need some scrutiny, and I don't think this fairly straightforward PR is the place for doing something potentially risky.